src/components/VmDetails/cards/OverviewCard.js

```diff
@@ -1,47 +1,204 @@
+    subset[part] = vm.get(part)
+    if (subset[part].toJS && typeof subset[part].toJS === 'function') {
+      subset[part] = subset[part].toJS()
+    }
+  }
+  return subset
+}
 
 /**
  * Overview of the VM (icon, OS type, name, state, description)
+ *
+ * Edits:
+ *   - VmIcon?
+ *   - VM Name  -->  if the VM is running, a name change will go to 'next run'
+ *        TODO: The REST API return the current running value and flags "next_run_configuration_exists": true
+ *        TODO: and the next_run can be queried with ;next_run matrix param on Vm query (/vm/<vmId>;next_run)
+ *        TODO: therefore it is possible to highlight the individual fields that will change on next_run (i.e. VM shutdown)
+ *   - VM Description
  */
-const OverviewCard = ({ vm, onEditChange }) => {
-  const handleCheckbox = (e) => {
-    onEditChange(true, e.target.checked)
-  }
-
-  return (
-    <BaseCard
-      editTooltip={`Edit overview for ${vm.get('id')}`}
-      onStartEdit={() => { onEditChange(true) }}
-      onCancel={() => { onEditChange(false) }}
-      onSave={() => { onEditChange(false) }}
-    >
-      {({ isEditing }) => {
-        return (
-          <div>
-            <p className={style['demo-text']} style={{ marginRight: '25px' }}>
-              Overview content for {vm.get('name')} <br />
-            </p>
-
-            {isEditing && (
-              <div>
-                <p className={style['demo-text']}>EDITING 1</p>
-                <Checkbox onChange={handleCheckbox}>Check to mark the edit dirty.</Checkbox>
-              </div>
-            )}
-          </div>
-        )
-      }}
-    </BaseCard>
-  )
+class OverviewCard extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      vm: props.vm, // ImmutableJS Map
+      isEditing: false,
+      isDirty: false,
+      actionCorrelationId: null,
+      correlatedMessages: null,
+    }
+  }
+
+  static getDerivedStateFromProps (props, state) {
+    if (!state.isEditing) {
+      return { vm: props.vm }
+    }
+
+    // Check the results of the saveChanges call and either setup to drop out of
+    // edit mode, or pull the error messages to display on the card.
+    if (state.isEditing && state.actionCorrelationId && props.vm.hasIn(['actionResults', state.actionCorrelationId])) {
+      const actionResult = props.vm.getIn(['actionResults', state.actionCorrelationId])
+      if (actionResult) {
+        return { isEditing: false, isDirty: false, actionCorrelationId: null, correlatedMessages: null }
+      }
+      return {
+        correlatedMessages: props.userMessages.get('records').filter(
+          record => record.failedAction && record.failedAction.actionCorrelationId === state.actionCorrelationId
+        ),
+      }
+    }
+
+    return null
+  }
+
+  componentDidUpdate (prevProps, prevState) {
+    if (prevState.isEditing && !this.state.isEditing) {
+      this.props.onEditChange(false)
+    }
+  }
+
+  handleCardOnStartEdit = () => {
```

Comment:
I would like to avoid declaring methods like this, because:
1. It is still experimental and don't part ECMA specification
2. It is not consistent with rest of code
3. It is at odds with class-based style of OOP

Reply:
> 1. It is still experimental and don't part ECMA specification

It's stage 3 and that's why we have babel.

> 2. It is not consistent with rest of code

True.  Let's tag this one to go into the yet-to-be-created style guide.  Perhaps there is an es-lint rule that we can turn on to prohibit this style.

I'll change the style to match the rest of the code base.

> 3. It is at odds with class-based style of OOP

I'd never expect to extend one of our react components, nor have I seen that in patternfly-react.  So this isn't really a relevant point.